iroh-sync/src/sync.rs

```diff
@@ -100,6 +100,28 @@ pub struct SyncOutcome {
     pub num_sent: usize,
 }
 
+#[derive(Debug, Default)]
+struct Subscribers(Vec<flume::Sender<Event>>);
+impl Subscribers {
+    pub fn subscribe(&mut self, sender: flume::Sender<Event>) {
+        self.0.push(sender)
+    }
+    pub fn unsubscribe(&mut self, sender: &flume::Sender<Event>) {
+        self.0.retain(|s| !s.same_channel(sender));
+    }
+    pub fn send(&mut self, event: Event) {
+        self.0.retain(|sender| sender.send(event.clone()).is_ok())
```

Comment:
I don't think this should be `retain` unless I am missing sth

Reply:
Well, my idea was: `flume::Sender::send` only returns an error if the receiver is dropped. Through `retain`, all senders which are disconnected are removed. Wdyt?

Reply:
right, that makes sense, but it requires now always taking write lock..tradeoffs